src/Digdir.Domain.Dialogporten.Application/Common/Numbers/PersonIdentifier.cs

```diff
@@ -0,0 +1,18 @@
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Digdir.Domain.Dialogporten.Application.Common.Numbers;
+
+internal static partial class PersonIdentifier
```

Comment:
Denne burde gjøres mer generisk, og renames til `EndUserIdentifier`, hvor `person` er en type. Her vil det komme `systemuser` også. For en `person` vil det kunne være ulike `identifier-{iso3166-alpha2}`, hvor `no` vil ha ulike valideringsregler enn `dk` og `se` hvis vi skal støtte dem en vakker dag.

Noe ala dette (courtesy of chatgpt):
```c#
using System;
using System.Text.RegularExpressions;

static class EndUserId
{
    [GeneratedRegex(@"urn:altinn:([\w-]{5,20}):([\w-]{5,20})::([\w-]{5,30})", RegexOptions.None, matchTimeoutMilliseconds: 100)]
    private static readonly Regex idRegex;

    public static bool IsValid(string identifier)
    {
        var match = idRegex.Match(identifier);
        if (!match.Success) return false;

        string namespacePart = match.Groups[1].Value;
        string type = match.Groups[2].Value;
        string value = match.Groups[3].Value;

        return namespacePart switch
        {
            "person" => ValidatePerson(type, value),
            "systemuser" => ValidateSystemUser(type, value),
            _ => false
        };
    }

    private static bool ValidatePerson(string type, string value)
    {
        switch (type)
        {
            case "identifier-no": 
                return ValidateNorwegianIdentifier(value);
            default:
                return false;
        }
    }

    private static bool ValidateSystemUser(string type, string value)
    {
        return type == "uuid" && Guid.TryParse(value, out _);
    }

    private static bool ValidateNorwegianIdentifier(string value)
    {
        // Hent kode fra SocialSecurityNumber
    }
}

```

Reply:
Jeg skrev i en kommentar over at vi bare trenger å bry oss om `person` i første omgang, men https://github.com/Altinn/altinn-authorization/issues/657 spesifiserer også systemuser som er en annen type vi vet vi må støtte, så like greit å bare legge inn validering for den allerede nå siden den er såpass enkel. Her vil det på sikt antagelig komme en tredje variant, selv-registrerte brukere som ikke har dnr/fnr, men den er ikke avklart formen på ennå)

Reply:
Forslag implementert